perf/benchmark/runner/runner.py

```diff
@@ -117,42 +118,36 @@ def __init__(
         else:
             sys.exit("invalid mesh %s, must be istio or linkerd" % mesh)
 
-    def nosidecar(self, fortio_cmd):
-        basestr = "http://{svc}:{port}/echo?size={size}"
-        if self.mode == "grpc":
-            basestr = "-payload-size {size} {svc}:{port}"
-        return fortio_cmd + "_base " + basestr.format(
-            svc=self.server.ip, port=self.ports[self.mode]["direct_port"], size=self.size)
+    def get_protocol_uri_fragment(self):
+        return "https" if self.mode == "grpc" else "http" 
 
-    def serversidecar(self, fortio_cmd):
-        basestr = "http://{svc}:{port}/echo?size={size}"
-        if self.mode == "grpc":
-            basestr = "-payload-size {size} {svc}:{port}"
-        return fortio_cmd + "_serveronly " + basestr.format(
-            svc=self.server.ip, port=self.ports[self.mode]["port"], size=self.size)
+    def nosidecar(self):
+        basestr = "{protocol}://{svc}:{port}/"
```

Comment:
question: do we need payload size field here? I see you are passing the size as a fortio parameter, but do not use it here anymore.

Reply:
Yup, what about request size?

Reply:
If I understood it correctly, the code would use the same payload-size both for sizing both the request- and response- entity body. The request-body would only be set in `grpc` mode. This behavior should be intact:
With Nighthawk we the `Content-Length` request header, which will be understood by the load generator as a hint to add the specified number of bytes in there as a request body:

```
        if self.mode == "grpc":
            nighthawk_args.append("--h2")
            if self.size:
                nighthawk_args.append(
                    "--request-header \"content-length: {size}\"")
```